cmd/nerdctl/run_restart_linux_test.go

```diff
@@ -87,3 +89,48 @@ func TestRunRestart(t *testing.T) {
 	base.DumpDaemonLogs(10)
 	t.Fatalf("the container does not seem to be restarted")
 }
+
+func TestRunRestartWithOnFailure(t *testing.T) {
+	base := testutil.NewBase(t)
+	if base.Target == testutil.Nerdctl && os.Getenv("CONTAINERD_VERSION") != "main" {
+		t.Skip("only test containerd on main version")
+	}
+	tID := testutil.Identifier(t)
+	defer base.Cmd("rm", "-f", tID).Run()
+	base.Cmd("run", "-d", "--restart=on-failure:2", "--name", tID, testutil.AlpineImage, "sh", "-c", "exit 1").AssertOK()
+
+	check := func(log poll.LogT) poll.Result {
+		inspect := base.InspectContainer(tID)
+		if inspect.State != nil && inspect.State.Status == "exited" {
+			return poll.Success()
+		}
+		return poll.Continue("container is not yet exited")
+	}
+	poll.WaitOn(t, check, poll.WithDelay(100*time.Microsecond), poll.WithTimeout(60*time.Second))
+	inspect := base.InspectContainer(tID)
+	assert.Equal(t, inspect.RestartCount, 2)
+}
+
+func TestRunRestartWithUnlessStopped(t *testing.T) {
+	base := testutil.NewBase(t)
+	if base.Target == testutil.Nerdctl && os.Getenv("CONTAINERD_VERSION") != "main" {
+		t.Skip("only test containerd on main version")
+	}
```

Comment:
Could you use `testutil.RequireContainerdPlugin()` ? (being added in https://github.com/containerd/nerdctl/pull/726)